### PR TITLE
Add support for X509 certificates

### PR DESCRIPTION
**Pull request by JacobBarthelmeh**

### CODE REVIEW

**Comment by JacobBarthelmeh** [comment]
retest this

**Comment by JacobBarthelmeh** [comment]
interop testing with PKIXSSH

Download pkixssh (https://roumenpetrov.info/secsh/)

Create pkixssh expected key format from john-cert.pem and john-key.pem using

```
cp ./keys/john-key.pem id_john
chmod 600 id_john
openssl x509 -in ./keys/john-cert.pem -subject -issuer -alias >> id_john
# use the pkixssh ssh-keygen tool
./ssh-keygen -y -f id_john > id_john.pub
```

Then connect to the example server using:

```
./examples/echoserver/echoserver -a ./keys/ca-ecc-cert.pem -J john:./keys/john-cert.pem

# using the pkixssh 'ssh'
./ssh -p 22222 -i id_john john@127.0.0.1
```

File: README.md [hunk 1]
```diff
@@ -462,3 +468,24 @@ NOTE: when prompted, enter the password which is "upthehill".
...
+
+    $ ./configure --enable-certs
+    $ make
+
+To provide a CA root certificate to validate a user's certificate, give the
+echoserver the command line option `-a`.
+
+    $ ./examples/echoserver/echoserver -a ./keys/ca-cert.pem
+
+The echoserver and client have a fake user named "john" whose certificate
```
Comment (dgarske) [inline README.md:490]: `echoserver and echoclient` or maybe `each server and client`?

File: README.md [hunk 2]
```diff
@@ -462,3 +468,24 @@ NOTE: when prompted, enter the password which is "upthehill".
...
+To compile wolfSSH with X.509 support, use the `--enable-certs` build option
+or define `WOLFSSH_CERTS`:
+
+    $ ./configure --enable-certs
+    $ make
+
+To provide a CA root certificate to validate a user's certificate, give the
+echoserver the command line option `-a`.
+
+    $ ./examples/echoserver/echoserver -a ./keys/ca-cert.pem
```
Comment (dgarske) [inline README.md:488]:
```% ./examples/echoserver/echoserver -a ./keys/ca-cert.pem
Couldn't find size of file ./keys/ca-cert.pem.```
Seems file is missing or wrong file is used here? Is it supposed to be keys/ca-ecc-cert.pem?

File: examples/client/client.c [hunk 3]
```diff
@@ -198,6 +204,15 @@ static word32 userPrivateKeyTypeSz = 0;
 static byte isPrivate = 0;
 
 
+#ifdef WOLFSSH_CERTS
+#if 0
```
Comment (dgarske) [inline examples/client/client.c:208]: If have an `#if 0` section here please add inline code comment explaining difference. Like RSA vs ECC.

File: README.md [hunk 4]
```diff
@@ -462,3 +468,24 @@ NOTE: when prompted, enter the password which is "upthehill".
...
+
```
Comment (dgarske) [inline README.md:489]: What is an example for the client command to test this?

**Review by dgarske — CHANGES_REQUESTED** [review]

**Review by dgarske — APPROVED** [review]

**Pull request merged**